packages/uix-core/src/tunnel/tunnel.ts

```diff
@@ -221,21 +257,28 @@ export class Tunnel extends EventEmitter {
     };
 
     timeout = window.setTimeout(() => {
-      tunnel.emitLocal(
-        "error",
+      tunnel.abort(
         new Error(
           `Timed out waiting for initial response from parent after ${config.timeout}ms`
         )
       );
-      tunnel.destroy();
     }, config.timeout);
 
     window.addEventListener("message", acceptListener);
-    tunnel.on("destroyed", cleanup);
-    tunnel.on("connected", cleanup);
+    tunnel.on("destroyed", () => {
+      cleanup();
+    });
+    tunnel.on("connected", () => {
+      cleanup();
+    });
 
-    const sendOffer = () =>
-      source.postMessage(messenger.makeOffered(key), config.targetOrigin);
+    const sendOffer = () => {
+      if (tunnel.isConnected) {
+        clearInterval(retrying);
+      } else {
+        source.postMessage(messenger.makeOffered(key), config.targetOrigin);
+      }
+    };
```

Comment:
This was a serious connection issue; the retry interval is short, and it fired several times after connection was complete, which would trigger the parent frame to discard the old connection and try to accept the new one.